include/deal.II/fe/fe_values.h

```diff
@@ -1663,9 +1687,9 @@ class FEValuesBase : protected FEValuesData<dim,spacedim>,
    *
    * @dealiiRequiresUpdateFlags{update_values}
    */
-  template <class InputVector, typename number>
+  template <class InputVector>
   void get_function_values (const InputVector &fe_function,
-                            std::vector<number> &values) const;
+                            std::vector<typename InputVector::value_type> &values) const;
```

Comment:
Here and below (in `FEValuesBase`) we should really be using `ProductType<double,typename InputVector::value_type>::type` as well. I'm willing to leave this as a follow-up patch, though, if you want, in the interest of moving things forward. Any thoughts?